### PR TITLE
Issue/run id testrail via config

### PR DESCRIPTION
**Pull request by OliverHoenig**
Added getRunId() which implements the behavior that the runId set in the config-file is taken into account and is preferred to the runId set in the annotation.

### CODE REVIEW

File: qtaf-testrail-plugin/src/main/java/de/qytera/qtaf/testrail/event_subscriber/UploadTestsSubscriber.java [hunk 1]
```diff
@@ -168,5 +169,35 @@ public void handleScenarioFailure(TestScenarioLogCollection scenarioLog, TestRai
...
+        String runId;
+
+        if (null != config.getString("testrail.runId")){
+            runId = config.getString("testrail.runId");
+            return runId;
+        }
+
+        if (null == testRailIdAnnotation){
+            throw new NullPointerException("The passed testRailId annotation is null");
+        }
```
Comment (qytera-development) [inline qtaf-testrail-plugin/src/main/java/de/qytera/qtaf/testrail/event_subscriber/UploadTestsSubscriber.java:192]: Very nice!

File: qtaf-testrail-plugin/src/test/java/de/qytera/testrail/event_subscriber/UploadTestsSubscriberTest.java [hunk 2]
```diff
@@ -44,6 +45,26 @@ public void testInitialization() {
...
+
+        Assert.assertThrows(NullPointerException.class, ()->subscriber.getRunId(null));
+
+        TestRail runIdEmptyAnnotation = dummyClass.getMethod("testDummyRunIdEmptyAnnotation").getAnnotation(TestRail.class);
+        Assert.assertThrows(IllegalArgumentException.class, () -> subscriber.getRunId(runIdEmptyAnnotation));
+
+        TestRail runIdAnnotatedAnnotation = dummyClass.getMethod("testDummyRunIdAnnotated").getAnnotation(TestRail.class);
+        Assert.assertEquals(subscriber.getRunId(runIdAnnotatedAnnotation), "01");
+    }
+
```
Comment (qytera-development) [inline qtaf-testrail-plugin/src/test/java/de/qytera/testrail/event_subscriber/UploadTestsSubscriberTest.java:67]: This test case is totally correct, but I would split it up into three test cases, for each assertion one test case. This makes it easier to find errors later if a test case fails.

**Review by qytera-development — CHANGES_REQUESTED** [review]
The code looks very good. You should split up the test case into three test cases, because we are testing three differenmt scenarios here:
- no run id given
- empty run id given
- correct run id given
This makes identification of an error easier later.

File: qtaf-testrail-plugin/src/main/java/de/qytera/qtaf/testrail/event_subscriber/UploadTestsSubscriber.java (hunk 1 above)
Comment (OliverHoenig) [reply qtaf-testrail-plugin/src/main/java/de/qytera/qtaf/testrail/event_subscriber/UploadTestsSubscriber.java:192]: Thanks :)

File: qtaf-testrail-plugin/src/test/java/de/qytera/testrail/event_subscriber/UploadTestsSubscriberTest.java (hunk 2 above)
Comment (OliverHoenig) [reply qtaf-testrail-plugin/src/test/java/de/qytera/testrail/event_subscriber/UploadTestsSubscriberTest.java:76]: Yes that is understandable. I have adjusted it.

**Review by qytera-development — APPROVED** [review]
That looks good

**Pull request merged**